src/lib/builtins/array.rs

```diff
@@ -472,6 +472,54 @@ pub fn find_index(this: &Value, args: &[Value], interpreter: &mut Interpreter) -
     Ok(Gc::new(ValueData::Number(f64::from(-1))))
 }
 
+/// Array.prototype.fill ( value[, start[, end]] )
+///
+/// The method fills (modifies) all the elements of an array from start index (default 0)
+/// to an end index (default array length) with a static value. It returns the modified array
+/// <https://tc39.es/ecma262/#sec-array.prototype.fill>
+pub fn fill(this: &Value, args: &[Value], _: &mut Interpreter) -> ResultValue {
+    let len: i32 = from_value(this.get_field_slice("length")).expect("Could not get argument");
+    let default_value = Gc::new(ValueData::Undefined);
+    let value = args.get(0).unwrap_or(&default_value);
+    let relateive_start: i32 = match args.get(1) {
+        Some(val) => {
+            if val.is_string() {
+                val.to_num() as i32
+            } else {
+                val.to_int()
```

Comment:
@letmutx I had problem here using only `val.to_int()` - it was converting every float to zero (e.g. `'2.5'` -> `0`)
So I had to introduce this `val.to_num() as i32` conversion

Reply:
The following seems to work. 
```
diff --git a/src/lib/builtins/array.rs b/src/lib/builtins/array.rs
index fe3a4aa..1e6caa8 100644
--- a/src/lib/builtins/array.rs
+++ b/src/lib/builtins/array.rs
@@ -482,23 +482,11 @@ pub fn fill(this: &Value, args: &[Value], _: &mut Interpreter) -> ResultValue {
     let default_value = Gc::new(ValueData::Undefined);
     let value = args.get(0).unwrap_or(&default_value);
     let relateive_start: i32 = match args.get(1) {
-        Some(val) => {
-            if val.is_string() {
-                val.to_num() as i32
-            } else {
-                val.to_int()
-            }
-        }
+        Some(val) => val.to_num() as i32,
         None => 0,
     };
     let relative_end: i32 = match args.get(2) {
-        Some(val) => {
-            if val.is_string() {
-                val.to_num() as i32
-            } else {
-                val.to_int()
-            }
-        }
+        Some(val) => val.to_num() as i32,
         None => len,
     };
     let k = if relateive_start < 0 {
```

Reply:
I saw that `val.to_num() as i32` is converting `ValueData::Undefined` to a very small negative number `-2147483648`.
But seems like the tests are all succeeding. 

I'm going to add this change

Reply:
`undefined` should behave like `None` case. Taken from Firefox
```
var a = [1, 2, 3];
a.fill(4, undefined, undefined);
> Array(3) [ 4, 4, 4 ]
​```

Reply:
@letmutx thanks for discovering these issues.
I updated updated the code and added more tests. I also noticed that `NaN` is still not implemented so I commented out tests using that value